cpp/src/geometry/box.cpp

```diff
@@ -20,7 +20,15 @@ Box::Box(const double &x_min, const double &y_min, const double &x_max, const do
       z_max(std::numeric_limits<double>::max())
 {
     if (x_max < x_min || y_max < y_min)
-        throw std::runtime_error("One or more of min values is greater than a value");
+        throw std::runtime_error("One or more of min values is greater than a max value");
+}
+
+// Vector3 constructor
+Box::Box(const Vector3 &min, const Vector3 &max)
+    : x_min(min.x), y_min(min.y), z_min(min.z), x_max(max.x), y_max(max.y), z_max(max.z)
+{
+    if (x_max < x_min || y_max < y_min || z_max < z_min)
+        throw std::runtime_error("One or more of min values is greater than a max value");
```

Comment:
all these constructors you can chain to a base constructor that does this check to clean up the code:

```
Box::Box(const Vector3 &min, const Vector3 &max)
    : Box(min.x, min.y, min.z, max.x, max.y, max.z) {};
```

Reply:
Nice one!